lib/adal-angular.js

```diff
@@ -328,33 +311,40 @@
                         // Loading with injector is suggested at github. https://github.com/angular/angular.js/issues/2367
 
                         config.headers = config.headers || {};
-
                         var resource = authService.getResourceForEndpoint(config.url);
                         authService.verbose('Url: ' + config.url + ' maps to resource: ' + resource);
                         if (resource === null) {
                             return config;
                         }
-
                         var tokenStored = authService.getCachedToken(resource);
                         if (tokenStored) {
                             authService.info('Token is available for this url ' + config.url);
                             // check endpoint mapping if provided
                             config.headers.Authorization = 'Bearer ' + tokenStored;
                             return config;
-                        } else {
+                        }
+                        else {
                             // Cancel request if login is starting
                             if (authService.loginInProgress()) {
-                                authService.info('login already started.');
-                                return $q.reject('login in progress, cancelling the request');
-                            } else {
+                                authService.info('login is in progress.');
+                                config.data = 'login in progress, cancelling the request';
```

Comment:
Nit: we should be consistent in our messages: either capitalize the first word for all of them, or have them all lowercase. Below, "Error" is capitalized.